src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -9,34 +11,216 @@
     create_csv_document,
     create_report_document,
 )
+from openforms.registrations.exceptions import RegistrationFailed
 from openforms.submissions.exports import create_submission_export
 from openforms.submissions.mapping import SKIP, FieldConf, apply_data_mapping
 from openforms.submissions.models import Submission, SubmissionReport
 from openforms.variables.utils import get_variables_for_context
 
 from ...constants import REGISTRATION_ATTRIBUTE, RegistrationAttribute
 from .client import DocumentenClient, get_documents_client
-from .typing import ConfigVersion, RegistrationOptionsV1, RegistrationOptionsV2
+from .models import ObjectsAPIRegistrationData
+from .typing import (
+    ConfigVersion,
+    RegistrationOptions,
+    RegistrationOptionsV1,
+    RegistrationOptionsV2,
+)
+
+
+def build_options(
+    plugin_options: RegistrationOptions, key_mapping: dict[str, str]
+) -> dict[str, Any]:
+    """
+    Construct options from plugin options dict, allowing renaming of keys
+    """
+    options = {
+        new_key: plugin_options[key_in_opts]
+        for new_key, key_in_opts in key_mapping.items()
+        if key_in_opts in plugin_options
+    }
+    return options
+
+
+def registration_submission_pdf(
```

Comment:
can you make the function name consistent? `registration_submission_pdf` vs. `register_submission_csv` - please use the same verb :)